### DISCUSSION TITLE
Does Kaml no longer support Java8

### DISCUSSION THREAD

**Discussion by 4o4E** [charleskorn/kaml#580] [Q&A] [answered]
I encountered an issue where my application could no longer run on Java 8 after updating Kotlin, Kotlinx serialization, and Kaml. Upon inspection, it was found that the dependency used was compiled using a higher version of Java (snakeyaml engine kmp 3.0.0)

`java.lang.UnsupportedClassVersionError: it/krzeminski/snakeyaml/engine/kmp/exceptions/MarkedYamlEngineException has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0`

**Comment by charleskorn** [comment, marked as answer]
It looks like that issue is coming from `snakeyaml-engine-kmp`, not kaml.

@krzema12 could you please take a look at this? You might need to add something like [this](https://github.com/charleskorn/kaml/blob/59a42fe3e94a519ae48250ea7f2dd539d699a32b/build.gradle.kts#L98) to your Gradle config.

**Reply by krzema12** [reply]
Sure thing, tracking in https://github.com/krzema12/snakeyaml-engine-kmp/issues/202

**Reply by krzema12** [reply]
snakeyaml-engine-kmp now supports Java 8: https://github.com/krzema12/snakeyaml-engine-kmp/releases/tag/v3.0.3
